WELL-ARCHITECTED.md

```diff
@@ -0,0 +1,210 @@
+* [x] **[Best]** Smaller functions require fewer permissions
+* [x] **[Required]** Use temporary credentials between resources and components
+
+* [ ] None of these
+
+
+##### Notes
+
+> * We use IAM policy to ensure that resources can only be called by other resources that should be calling them.
+>
+> * The pipeline will assume a role with only the permissions it requires to deploy the application. This will either be only being able to perform a specific action on multiple resources or any action on a particular resource. 
+>
+> * This application does not use private networking. 
+>
+> * We have individual functions for each different piece of business logic. 
+
+---
+
+#### SEC 3: How do you implement Application Security in your workload?***
+
+* [ ] Question does not apply to this workload
+
+
+* [x] **[Required]** Review security awareness documents frequently
+* [x] **[Required]** Store secrets that are used in your code securely
+* [ ] **[Good]** Implement runtime protection to help prevent against malicious code execution
+* [ ] **[Best]** Automatically review workload’s code dependencies/libraries
+* [x] **[Best]** Validate inbound events
+
+
+* [ ] None of these
+
+
+##### Notes
+
+> * This application doesn't have any stored secrets. The GitHub token is required by CodePipeline, this is passed as a string for CloudFormation, it is not however visible within the CloudFormation console. This could be improved by manually creating a secrets manager entry for the token and replacing the CloudFormation parameter for the token with the secrets manager value by utilising Dynamic References.
+>https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html
+>
+> * For reviewing dependencies and libraries we could integrate an automatic check into the pipeline. There are many tools and providers which can check code. Currently this is manual using PEP8 and Bandit manual checks. 
+>
+> * We only check for particular events and check to make sure the object is valid. 
+
+---
+
+## Reliability
+
+#### REL 1. How do you regulate inbound request rates?
+
+* [ ] Question does not apply to this workload
+
+
+* [x] **[Required]** Use throttling to control inbound request rates
+* [ ] **[Good]** Use, analyze and enforce API quotas
+* [X] **[Best]** Use mechanisms to protect non-scalable resources
+
+
+* [ ] None of these
+
+
+##### Notes
+
+> * We are using SQS queues in front of our Lambda functions, this helps us throttle the rate at which our application processes requests. 
+>
+> * We don't have API's to set quotas for.
+>
+> * Our downstream resources are S3 and DynamoDB on-demand which are more than capable of scaling to match our volumes.
+
+---
+
+#### REL 2. How do you build resiliency into your Serverless application?
+
+* [ ] Question does not apply to this workload
+
+
+* [x] **[Required]** Manage transaction, partial, and intermittent failures
+* [ ] **[Required]** Manage duplicate and unwanted events
```

Comment:
I think we have this requirement met.  We handle duplicates well - the Dynamo key and converted S3 object for each analysis is tied to the input object being analyzed.  Pushing the same document will result in the same artifact.  There could be a question regarding document names (what happens when a new document with the same name) but I think we check this box.

Reply:
Checked and updated supporting text.